packages/big-design/src/components/Checkbox/Checkbox.tsx

```diff
@@ -58,6 +66,14 @@ const RawCheckbox: React.FC<CheckboxProps & PrivateProps> = ({
     warning('label must be either a string or a CheckboxLabel component.');
   }, [disabled, hiddenLabel, id, label, labelId]);
 
+  const renderedDescription = useMemo(() => {
+    if (!description) {
+      return null;
+    }
+
+    return <Description link={description.link}>{description.text}</Description>;
+  }, [description]);
```

Comment:
I can see this always rerendering this memo since description is a value by reference. To prevent that, you could possibly modify it like so:
```suggestion
  }, [description.link, description.text]);
```